drivers/input/input_ft5336.c

```diff
@@ -77,35 +84,46 @@ static int ft5336_process(const struct device *dev)
 	int r;
 	uint8_t points;
 	uint8_t coords[4U];
-	uint8_t event;
 	uint16_t row, col;
 	bool pressed;
 
-	/* obtain number of touch points (NOTE: multi-touch ignored) */
+	/* obtain number of touch points */
 	r = i2c_reg_read_byte_dt(&config->bus, REG_TD_STATUS, &points);
 	if (r < 0) {
 		return r;
 	}
 
-	points = (points >> TOUCH_POINTS_POS) & TOUCH_POINTS_MSK;
-	if (points != 0U && points != 1U) {
-		return 0;
-	}
+	points = FIELD_GET(TOUCH_POINTS_MSK, points);
+	if (points != 0) {
+		/* Any number of touches still counts as one touch. All touch
+		 * points except the first are ignored. Obtain first point
+		 * X, Y coordinates from:
+		 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
+		 * We ignore the Event Flag because Zephyr only cares about
+		 * pressed / not pressed and not press down / lift up
+		 */
+		r = i2c_burst_read_dt(&config->bus, REG_P1_XH, coords, sizeof(coords));
+		if (r < 0) {
+			return r;
+		}
 
-	/* obtain first point X, Y coordinates and event from:
-	 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
-	 */
-	r = i2c_burst_read_dt(&config->bus, REG_P1_XH, coords, sizeof(coords));
-	if (r < 0) {
-		return r;
-	}
+		row = ((coords[0] & POSITION_H_MSK) << 8U) | coords[1];
+		col = ((coords[2] & POSITION_H_MSK) << 8U) | coords[3];
 
-	event = (coords[0] >> EVENT_POS) & EVENT_MSK;
-	row = ((coords[0] & POSITION_H_MSK) << 8U) | coords[1];
-	col = ((coords[2] & POSITION_H_MSK) << 8U) | coords[3];
-	pressed = (event == EVENT_PRESS_DOWN) || (event == EVENT_CONTACT);
+		uint8_t touch_id = FIELD_GET(TOUCH_ID_MSK, coords[2]);
 
-	LOG_DBG("event: %d, row: %d, col: %d", event, row, col);
+		if (touch_id != TOUCH_ID_INVALID) {
+			pressed = true;
+			LOG_WRN("points: %d, touch_id: %d, row: %d, col: %d",
```

Comment:
isn't this the "ok" case and the other one should be a LOG_WRN instead?

Reply:
> isn't this the "ok" case and the other one should be a LOG_WRN instead?

Argh. Coding before coffee.